routes/libraries.js

```diff
@@ -18,188 +17,219 @@ router.route("/").get(async (req, res) => {
+  let address = '';
+
+  /** This axios request does reverse geocatching to get the address of the library */
+  let data = await axios.get(`https://maps.googleapis.com/maps/api/geocode/json?latlng=${newLibraryData.lat},${newLibraryData.lng}&key=AIzaSyAPxSPvWssw3gI4W1qJaFk9xlBqBicI3iY`);
+
+  if (!data.data.results || (data.data.results.length === 0) || (data.data.status === "ZERO_RESULTS")) throw "Error: Location not found for these co-ordinates";
+
+  city2 = data.data.results[5].address_components[0].long_name;
+  city = data.data.results[7].address_components[0].long_name;
+
+  if (city !== "Hoboken" && city2 === "Hoboken" || city2 === "07030") {
+    address = data.data.results[0].formatted_address;
+  } else {
+    address = data.data.results[1].formatted_address;
+  }
+
+  return { city, city2, address };
+}
+
+const handleValidationErrors = (res, req, action, errorField, error, newLibraryData) => {
+  let errorObj = {
+    title: "Creating a Library",
+    user: req.session.user,
+    editOrCreate: action,
+    hasErrors: true,
+    library: newLibraryData
+  };
+  errorObj[errorField] = error;
+
+  /** We don't have to let the user know if any error has occured while deleting the image */
+  if (req.file) fs.unlink(req.file.path, () => { });
+
+  return res.status(400).render("libraries/new", errorObj);
+}
+
+async function routeValidationsForLibrary(newLibraryData, res, req) {
+  let errors = [];
+
+  try {
+    newLibraryData.name = validation.checkString(newLibraryData.name, "Library name");
+  } catch (e) {
+    return handleValidationErrors(res, req, "Create", "nameError", e, newLibraryData);
+  }
+
+  try {
+    newLibraryData.lat = Number(newLibraryData.lat);
+    newLibraryData.lat = validation.isValidNumber(newLibraryData.lat, "Librarys Latitude");
+  } catch (error) {
+    return handleValidationErrors(res, req, "Create", "latError", e, newLibraryData);
+  }
+
+  try {
+    newLibraryData.lng = Number(newLibraryData.lng);
+    newLibraryData.lng = validation.isValidNumber(newLibraryData.lng, "Librarys Longitude");
+  } catch (error) {
+    return handleValidationErrors(res, req, "Create", "lngError", e, newLibraryData);
+  }
+
+  /** Something went wrong saving the image */
+  if (!req.file) return handleValidationErrors(res, req, "Create", "imageError", "Image is of invalid type or no image has been selected", newLibraryData);
+
+  let addressObj = {};
+
+  try {
+    addressObj = await reverseGeoCodeCoordinates(newLibraryData);
+  } catch (e) {
+    if (
+      (typeof e === "string") &&
+      (e === "Error: Location not found for these co-ordinates")
+    ) return handleValidationErrors(res, req, "Create", "latError", e, newLibraryData);
+
+    if (req.file) fs.unlink(req.file.path, () => { });
+
+    console.error(e);
+    return res.status(500).render('error', { errorNum: 500, title: "Error" });
+  }
+
+  let { city, city2, address } = addressObj;
+
+  if (city !== "Hoboken" && city2 !== "Hoboken" && city2 !== "07030") {
```

Comment:
This like should be
 if((city !== "Hoboken" && city2 !== "Hoboken" && city2 !== "07030") || address.includes("Hoboken")) {

Reply:
shouldn't it be !address.includes("Hoboken") since we are checking if the address doesn't have hoboken in it

Reply:
!address.includes("Hoboken") should fix the bug we were experiencing